### PR TITLE
DS Picker takes the max height available and flip when needed

### LINKED ISSUE #67897: Make the DS picker dropdown container to grow depending of the viewport height

### PR DESCRIPTION
**Pull request by ivanortegaalba**
**Problem**
Fixes #68323 
* The height of the dropdown was fixed to a certain amount of pixels. This doesn't allow to take advantage of the available height or it was provoking unnecessary flips.

**Solution**
* Compute the height avoiding overflows + margin + flipping when the min-height is reached

|Adjusting height|Flipping when needed|
|-|-|
|

### CODE REVIEW

File: public/app/features/datasources/components/picker/popperModifiers.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  name: 'applyMaxSize',
+  enabled: true,
+  phase: 'beforeWrite',
+  requires: ['maxSize'],
+  fn({ state }: ModifierArguments<{}>) {
+    const { height, width } = state.modifiersData.maxSize;
+    state.styles.popper.maxHeight = `${height - MODAL_MARGIN}px`;
+    state.styles.popper.minHeight = `${FLIP_THRESHOLD}px`;
+    state.styles.popper.maxWidth = width;
+    console.log('state.styles.popper.maxWidth', state.styles.popper.maxWidth);
```
Comment (juanicabanas) [inline public/app/features/datasources/components/picker/popperModifiers.ts:41]: should we remove this log?

**Review by juanicabanas — APPROVED** [review]
LGTM!

File: public/app/features/datasources/components/picker/popperModifiers.ts (hunk 1 above)
Comment (ivanortegaalba) [reply public/app/features/datasources/components/picker/popperModifiers.ts:41]: Ups 🙈

File: public/app/features/datasources/components/picker/popperModifiers.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const { width, height } = state.rects.popper;
+    const [basePlacement] = state.placement.split('-');
+
+    const widthProp = basePlacement === 'left' ? 'left' : 'right';
+    const heightProp = basePlacement === 'top' ? 'top' : 'bottom';
+    console.log('width', width);
+    console.log('height', height);
+    console.log('overflow', overflow);
+    console.log('x', x);
+    console.log('y', y);
```
Comment (dprokop) [inline public/app/features/datasources/components/picker/popperModifiers.ts:23]:
```suggestion
```

File: public/app/features/datasources/components/picker/popperModifiers.ts (hunk 1 above)
Comment (dprokop) [inline public/app/features/datasources/components/picker/popperModifiers.ts:41]:
```suggestion
```

**Pull request merged**